File: src_a_(C++)/Test5-10.c
//Ïðîåêò Ñòðàííèê-Ìîäóëà Äëÿ Windows 32, òåñòîâàÿ ïðîãðàììà
//Ãðóïïà òåñòîâ 5:ÏÐÎÖÅÄÓÐÛ
//Òåñò íîìåð    10:ÊÎÑÂÅÍÍÀß ÐÅÊÓÐÑÈß

include Win32

char s[15];

int i;

void pr1(int j);
void pr2(int j);

void pr1(int j) {
  if(j>0) pr2(j-1);
  else return;
  i=i+1;
}

void pr2(int j) {
  if(j>0) pr1(j-1);
  else return;
  i=i+1;
}

void main() {
  i=4;
  pr1(8);
  wvsprintf(s,"i=%li",&i);
  MessageBox(0,s,"i=12",0);
}

File: src_a_(Pascal)/Test5-10.pas
//Ïğîåêò Ñòğàííèê-Ìîäóëà Äëÿ Windows 32, òåñòîâàÿ ïğîãğàììà
//Ãğóïïà òåñòîâ 5:ÏĞÎÖÅÄÓĞÛ
//Òåñò íîìåğ    10:ÊÎÑÂÅÍÍÀß ĞÅÊÓĞÑÈß
program Test5_10;
uses Win32;

var s:string[15];

var i:integer;

procedure pr1(j:integer); forward; 
procedure pr2(j:integer); forward;

procedure pr1(j:integer);
begin
  if j>0
    then pr2(j-1)
    else return;
  i:=i+1
end;

procedure pr2(j:integer);
begin
  if j>0
    then pr1(j-1)
    else return;
  i:=i+1
end;

begin
  i:=4;
  pr1(8);
  wvsprintf(s,'i=%li',addr(i));
  MessageBox(0,s,'i=12',0);
end.

